From: Egbert Eich <eich@suse.com>
Date: Mon, 4 Mar 2013 09:24:38 -0500
Subject: [PATCH] DRM/i915: On G45 enable cursor plane briefly after enabling the display plane.
Patch-mainline: Never
References: bnc#918161
Signed-off-by: Egbert Eich <eich@suse.com>

drm/i915: On VLV enable cursor plane briefly after enabling the display plane

VLV has the same issue G45 has: the overlay enable doesn't work
under certain circumstances.
We used to work around this with patch:
    commit 61bc95c1fbbb6a08b55bbe161fdf1ea5493fc595
    Author: Egbert Eich <eich@suse.com>
    Date:   Mon Mar 4 09:24:38 2013 -0500

    DRM/i915: On G45 enable cursor plane briefly after enabling the display plane.

(See also bnc#753371)

This problem has meanwhile been understood:
   commit 564ed191f5d816d24501664296991ec70327e2bc
   Author: Imre Deak <imre.deak@intel.com>
   Date:   Fri Jun 13 14:54:21 2014 +0300

    drm/i915: gmch: fix stuck primary plane due to memory self-refresh mode

thus 61bc95c1fbbb6a08b55bbe161fdf1ea5493fc595 could be removed.

Backporting 564ed191f5d816d24501664296991ec70327e2bc however requires to backport
a number of other patches as well. We therefore settle with commit 61bc95c1fbbb6a
and extend it to VLV.

---
 drivers/gpu/drm/i915/intel_display.c |    2 ++
 1 file changed, 2 insertions(+)

--- linux-3.12-SLE12.orig/drivers/gpu/drm/i915/intel_display.c
+++ linux-3.12-SLE12/drivers/gpu/drm/i915/intel_display.c
@@ -3734,6 +3734,8 @@ static void valleyview_crtc_enable(struc
 	intel_enable_pipe(dev_priv, pipe, false);
 	intel_enable_plane(dev_priv, plane, pipe);
 	intel_enable_planes(crtc);
+	/* The fixup needs to happen before cursor is enabled */
+	g4x_fixup_plane(dev_priv, pipe);
 	intel_crtc_update_cursor(crtc, true);
 
 	intel_update_fbc(dev);
